src/proxy/inapp/inapp_full_proxy.go

```diff
@@ -0,0 +1,188 @@
+package app
+
+import (
+	"fmt"
+	"time"
+
+	bcrypto "github.com/mosaicnetworks/babble/src/crypto"
+	hg "github.com/mosaicnetworks/babble/src/hashgraph"
+	bproxy "github.com/mosaicnetworks/babble/src/proxy/babble"
+	"github.com/sirupsen/logrus"
+)
+
+//InmemFullProxy serves as an Inapp proxy for those whom
+type InmemFullProxy struct {
+	submitCh              chan []byte
+	commitCh              chan bproxy.Commit
+	stateHash             []byte
+	committedTransactions [][]byte
+	logger                *logrus.Logger
+	snapshotRequestCh     chan bproxy.SnapshotRequest
+	restoreCh             chan bproxy.RestoreRequest
+	timeout               time.Duration
+}
+
+// NewInmemFullProxy instantiate an InmemFullProxy.
+// If no logger, a new one is created
+func NewInmemFullProxy(timeout time.Duration, logger *logrus.Logger) *InmemFullProxy {
+	if logger == nil {
```

Comment:
How does one provide/retrieve the CommitCh or SubmitCh?
Actually, for mobile bindings (Java, ObjectiveC), we cannot use go channels.
We need to use callbacks. Check out the mobile app proxy.

Reply:
Fixed